packages/actor-http-fetch/lib/ActorHttpFetch.ts

```diff
@@ -76,12 +76,46 @@ export class ActorHttpFetch extends ActorHttp {
 
     try {
       requestInit = await this.fetchInitPreprocessor.handle(requestInit);
+      let retryCount: number = action.context?.get(KeysHttp.httpRetryCount) ?? 0;
 
-      // Perform request
       const customFetch: ((input: RequestInfo, init?: RequestInit) => Promise<Response>) | undefined = action
         .context?.get(KeysHttp.fetch);
 
-      const response = await (customFetch || fetch)(action.input, requestInit);
+      // Actual fetch respecting timeout and retry count.
+      const getResponse = async(): Promise<Response> => {
+        // When retry count is greater than 0, record the errors for failed requests.
+        const errors: unknown[] = [];
+        while (retryCount-- >= 0) {
+          try {
+            return await (customFetch || fetch)(action.input, requestInit);
+          } catch (error: unknown) {
+            // If the fetch was aborted by timeout, we won't retry.
+            if (requestInit.signal?.aborted) {
+              throw error;
+            }
+            errors.push(error);
+            // Wait for specified delay.
+            if (retryCount >= 0) {
+              await new Promise((resolve, reject) => {
+                // Cancel waiting, if timeout is reached.
+                requestInit.signal?.addEventListener('abort', () => {
+                  reject(new Error('Fetch aborted by timeout.'));
+                });
+                setTimeout(resolve, action.context?.get(KeysHttp.httpRetryDelay) ?? 0);
+              });
+            }
+          }
+        }
+        // If retry count was 0, i.e. only one error was recorded, just throw it.
+        if (errors.length === 1) {
+          throw errors[0];
+        } else {
+          throw new Error(`Number of fetch retries exceeded. Encountered errors: ["${errors.join('", "')}"]`);
+        }
+      };
```

Comment:
Rather than reporting all the errors I would just report the first error encountered. I think it is fairly safe to assume that the same problem is going to be encountered each time and reporting all the errors is going to create massive error messages if you do a significant number of retries.

Reply:
I agree with the above.
We could optionally append something like `(failed for x retries)` to the error message.